src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java

```diff
@@ -104,10 +114,15 @@ public void updateHearingRequest(Long hearingId, UpdateHearingRequest hearingReq
      * @return HearingRequest HearingRequest
      */
     @Override
-    public HearingRequest validateGetHearingsRequest(String caseRef, String status) {
+    public GetHearingsResponse getHearings(String caseRef, String status) {
         log.info("caseRef:{} ; status:{}", caseRef, status);
-        // TODO: select hearing request from given caseRefId and status (if any)
-        return new HearingRequest();
+        List<CaseHearingRequestEntity> entities;
+        if (!isBlank(status)) {
+            entities = caseHearingRequestRepository.getHearingDetailsWithStatus(caseRef, status);
+        } else {
+            entities = caseHearingRequestRepository.getHearingDetails(caseRef);
+        }
+        return getHearingsResponseDetails(caseRef,entities);
```

Comment:
To avoid this fairly general class getting too bloated, might be worth moving the processing to a separate component?
```suggestion
        return hearingsResponseDetailsMapper.toResponse(caseRef, entities);
```

Reply:
refactored to a separate component.